### PR TITLE
Batch remove entries

### PR DESCRIPTION
**Pull request by NorwayMaple**
I changed the SQL in `DBMSProcessor` so that multiple entries are deleted with one query instead of using a query for each entry. This was the original rationale for the refactoring I did in October.

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

### CODE REVIEW

File: src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java [hunk 1]
```diff
@@ -164,18 +164,46 @@ void testUpdateEqualEntry(DBMSType dbmsType, DBMSConnection dbmsConnection, DBMS
 
     @ParameterizedTest
     @MethodSource("org.jabref.logic.shared.TestManager#getTestingDatabaseSystems")
-    void testRemoveEntry(DBMSType dbmsType, DBMSConnection dbmsConnection, DBMSProcessor dbmsProcessor) throws SQLException {
```
Comment (koppor) [inline src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java]: Can be keep the single test? -> Meaning the additional test for two entries is OK, but we should also keep the simple (and probably always succeeding test). - Reason: The for loop above is not entered for one entry and thus this test case is special.

Comment (koppor) [inline src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java]: Can be keep the single test? -> Meaning the additional test for two entries is OK, but we should also keep the simple (and probably always succeeding test). - Reason: The for loop above is not entered for one entry and thus this test case is special.

**Comment by koppor** [comment]
Minor thing: checkstyle should be fixed 😇

File: src/main/java/org/jabref/logic/shared/DBMSProcessor.java [hunk 2]
```diff
@@ -401,16 +401,22 @@ private void insertOrUpdateFields(BibEntry localBibEntry) throws SQLException {
...
+    public void removeEntries(List<BibEntry> bibEntries) {
         StringBuilder query = new StringBuilder()
                 .append("DELETE FROM ")
                 .append(escape("ENTRY"))
                 .append(" WHERE ")
                 .append(escape("SHARED_ID"))
-                .append(" = ?");
+                .append(" IN (");
+        for (int i = 0; i < bibEntries.size() - 1; i++) {
+            query.append("?, ");
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/logic/shared/DBMSProcessor.java]: I think you can use `"?, "`.repeat(bibEntries.size())` here

File: src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java (hunk 1 above)
Comment (NorwayMaple) [reply src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java]: I could, but it would still have to be `dbmsProcessor.removeEntries(Collections.singletonList(entryToRemove))` because I got rid of `DBMSProcessor.removeEntry` in the singular. Is it still worth it?

File: src/main/java/org/jabref/logic/shared/DBMSProcessor.java (hunk 2 above)
Comment (NorwayMaple) [reply src/main/java/org/jabref/logic/shared/DBMSProcessor.java]: Made the change, will push soon.

File: src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java (hunk 1 above)
Comment (koppor) [reply src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java]: Sure. The tests should be independent of the implemention somehow. Tests should cover the general case, but also edge cases, null list, 0 element list, list with 1 elements, list with integer.max elements (which is impossible IMHO), but the other cases should be easy to do?

Comment (NorwayMaple) [reply src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java]: Okay, in that case I will use `Collections.singletonList`. Should I add a test for each of those edge cases as well?

Comment (koppor) [reply src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java]: Sure, please go ahead. In case tests are easy to write, they should be written. - The thing is: The implementation could change in 1 year (who knows) and the test will still cover that case.

Please also add a test for

- `dbmsProcessor.removeEntries(Collections.emptyList())`
- `dbmsProcessor.removeEntries(null)` (should throw a NPE)

Comment (NorwayMaple) [reply src/test/java/org/jabref/logic/shared/DBMSProcessorTest.java]: Throw on empty list? If so, throw what? I can't imagine why someone would want to call `removeEntries` on purpose on an empty list.

**Comment by koppor** [comment]
I merged to move ahead. Think, the software engineering discussion can be done later :)

> Because the unit tests are not only for the current version of the code. They are especially for all future versions and if somebody changes the code and does not know what s/he did, the test can fail and tell him that he did something wrong. So you can't say that this will never happen.

https://sqa.stackexchange.com/a/9009/28592

Thus, I argue for improper usage - who knows, what future code will bring.

**Comment by NorwayMaple** [comment]
Thanks! Is there any way I can fix this in my following PR?
